### ISSUE TITLE
Add action: executing shell command

### ISSUE THREAD

**Issue by garywill** [keymapperorg/KeyMapper#661]
**Developer TODO (don't remove)**
- [x] Add a checkbox for root mode
- [x] Add a test button when setting up that shows you the output, make sure it is scrollable.
- [x] Make it full screen and not a bottom sheet because there is potentially a lot to write/output.
- [x] Add option to use ADB that requires PRO mode. Should button to enable pro mode if it is disabled. Show a small note that this does not support streaming output when testing.

-------------------------

Hi developer(s), thank you for great KeyMapper!

**It would be nice to run a shell command when I trigger by a key/button.** 

----------------------

Also, I'm looking for a way to combine KeyMapper with https://github.com/renyuneyun/Easer . Easer can set a user-custom shell command as action, but it can't set a key/button as trigger.

https://github.com/renyuneyun/Easer/issues/376

**Comment by sds100** [comment]
Hello!
Does Easer allow you to create launcher shortcuts for actions? If so then maybe the App Shortcut action will work in Key Mapper.

**Comment by garywill** [comment]
Easer supports home screen widgets. Every widget has an label ( I believe that's this widget's id)
Then you can set tapping a widget with an id as trigger event.

But that widgets seem different than launcher shortcut.
They won't appear in Key Mapper's "App Shortcut" section.

**Comment by sds100** [comment]
Oh okay. Can you trigger Easer actions with Intents? There is an action in Key Mapper to send an intent

**Comment by garywill** [comment]
Easer supports receiving broadcast as trigger event. I'm not sure if that is "intent" ( I don't know a lot about Android's intent).

This is Easer's screenshot
![Screenshot_20210509-194255_Easer](https://user-images.githubusercontent.com/32130780/117571275-66817600-b0bd-11eb-8992-27030070ada9.png)

If it is "intent", could you give me some idea how to do ?

**Comment by sds100** [comment]
Yeah, a broadcast is a way for an app to send an intent to another app. In that case you'll want to create an Intent action in Key Mapper and choose the "broadcast receiver" target. Set the Intent action in Easer (in your screenshot) to be the same as in Key Mapper.

**Comment by victorbnl** [comment]
I didn't understand the thing with Easer but being able to start shell commands with gestures would be really useful

**Comment by victorbnl** [comment]
Here's a temporary workaround
1. Install Termux
2. Create a `.bashrc` with
```
alias a="your command"
```
3. Create the action "Open Termux" with a delay then of about 1000 to 1500 ms
4. Create the action "Touch the screen" twice and first touch the `a` key of the keyboard, and then enter

**Comment by sds100** [comment]
Yeah, that works. I'll still leave the shell command feature request open

**Comment by victorbnl** [comment]
Yes it would be really cool to be able to execute commands, I think that's the most important and basic thing to have in this kind of app. If you can execute commands you can literally do whatever you do even though it's not made in-app, like uninstalling an app, change a parameter...
